ntoskrnl/ps/i386/psvdm.c

```diff
@@ -0,0 +1,73 @@
+/*
+ * PROJECT:     ReactOS Kernel
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Virtual DOS Machine (VDM) support routines
+ * COPYRIGHT:   Copyright Stefan Ginsberg (stefan.ginsberg@reactos.org)
+ *              Copyright 2020 George Bișoc (george.bisoc@reactos.org)
+ */
+
+/* INCLUDES ******************************************************************/
+
+#include <ntoskrnl.h>
+#define NDEBUG
+#include <debug.h>
+
+/* GLOBALS *******************************************************************/
+
+ERESOURCE VdmIoListCreationResource;
+
+/* FUNCTIONS *****************************************************************/
+
+/**
+ * @brief
+ * Deletes Virtual DOS Machine (DOS) objects, given an Executive process (EPROCESS).
+ * 
+ * @param[in] Process
+ * A pointer to an Executive process.
+ * 
+ * @return
+ * Nothing.
+ * 
+ */
+VOID
+NTAPI
+PspDeleteVdmObjects(IN PEPROCESS Process)
+{
+    /* If there are no VDM objects, just exit */
+    if (Process->VdmObjects == NULL)
+    {
+        DPRINT1("PspDeleteVdmObjects(): No VDM objects could be found, exiting...\n");
+        return;
+    }
+
+    /* Otherwise return the quota it was taking up and free them */
+    PsReturnProcessNonPagedPoolQuota(Process->VdmObjects, sizeof(Process->VdmObjects));
+    ExFreePoolWithTag(Process->VdmObjects, '  eK');
+    Process->VdmObjects = NULL;
+
+    /* Remove the VDM resource from the system's resources list and get our process quota */
+    ExDeleteResourceLite(&VdmIoListCreationResource);
+    PsReturnProcessPagedPoolQuota(Process, sizeof(Process));
+    DPRINT1("PspDeleteVdmObjects(): Successfully deleted the VDM objects!\n");
```

Comment:
Why do you delete the global resource on per-process cleanup? And why return the quota for the process here? Non of this seems to make any sense.